x-pack/plugin/data-frame/src/main/java/org/elasticsearch/xpack/dataframe/transforms/pivot/SchemaUtil.java

```diff
@@ -75,6 +76,8 @@ public static void deduceMappings(final Client client,
                 ValuesSourceAggregationBuilder<?, ?> valueSourceAggregation = (ValuesSourceAggregationBuilder<?, ?>) agg;
                 aggregationSourceFieldNames.put(valueSourceAggregation.getName(), valueSourceAggregation.field());
                 aggregationTypes.put(valueSourceAggregation.getName(), valueSourceAggregation.getType());
+            } else if(agg instanceof ScriptedMetricAggregationBuilder) {
+                // do nothing
```

Comment:
this needs to put the fieldname/mapping (`dynamic`)  but only aggregationTypes and not aggregationSourceFieldNames

(I am not sure what happens further down, might require additional checks)